library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
@kha I don't think it works. The parser action `p` is producing a pre-term that needs to be elaborated. When we write `parse (list nat) p`, `p` must produce a `pre-term` that when elaborated has type `list nat`. 

I think the strong typing you want would only work for closed terms. For closed terms, `parse` would be able to convert a `vm_obj` that encodes a `list nat` into a pre-term that represents the same `list`.